src/backend/d3d12/RenderPipelineD3D12.cpp

```diff
@@ -120,12 +121,14 @@ namespace d3d12 {
         descriptor.BlendState.RenderTarget[0].LogicOp = D3D12_LOGIC_OP_NOOP;
         descriptor.BlendState.RenderTarget[0].RenderTargetWriteMask = D3D12_COLOR_WRITE_ENABLE_ALL;
 
-        descriptor.DepthStencilState.DepthEnable = false;
-        descriptor.DepthStencilState.StencilEnable = false;
+        DepthStencilState* depthStencilState = ToBackend(GetDepthStencilState());
+        descriptor.DepthStencilState = depthStencilState->GetD3D12DepthStencilDescriptor();
+
         descriptor.SampleMask = UINT_MAX;
         descriptor.PrimitiveTopologyType = D3D12_PRIMITIVE_TOPOLOGY_TYPE_TRIANGLE;
         descriptor.NumRenderTargets = 1;
         descriptor.RTVFormats[0] = DXGI_FORMAT_R8G8B8A8_UNORM;
+        descriptor.DSVFormat = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
```

Comment:
Yes that's the interesting issue outlined above, we should gather the attachment format from the renderpass and subpass. At nxt::Framebuffer creation we will (or do already?) check that the framebuffer texture formats match the ones of the renderpass, so everything will line up correctly.